index.js

```diff
@@ -33,26 +37,111 @@ CredentialsAuthenticator.prototype.authenticate = function authenticate(plexApi,
         })
     };
 
-    request.post(options, function (err, res, xmlBody) {
-        var token;
+    return request.post(options)
+        .then(xmlBody => {
+            let token = extractAuthToken(xmlBody);
 
-        if (err) {
-            return callback(new Error('Error while requesting https://plex.tv for authentication: ' + String(err)));
-        }
-        if (res.statusCode !== 201) {
-            return callback(new Error('Invalid status code in authentication response from Plex.tv, expected 201 but got ' + res.statusCode));
-        }
+            if (!token) {
+                throw new Error('Couldnt not find authentication token in response from Plex.tv :(');
+            }
 
-        token = extractAuthToken(xmlBody);
-        if (!token) {
-            return callback(new Error('Couldnt not find authentication token in response from Plex.tv :('));
-        }
+            self.emit('authToken', token);
 
-        self.emit('token', token);
-        callback(null, token);
-    });
+            return token;
+        })
+        .catch(errors.StatusCodeError, err => {
+            throw new Error('Invalid status code in authentication response from Plex.tv, expected 201 but got ' + err.statusCode);
+        })
+        .catch(errors.RequestError, err => {
+            throw new Error('Error while requesting https://plex.tv for authentication: ' + String(err));
+        })
+        .then(token => {
+            if (self.homeUser && self.homeUser.name) {
+                let plexContext = {
+                    plexApi : plexApi,
+                    token   : token
+                };
+
+                return getHomeUser(plexContext)
+                    .then(xml  => findUserByName(xml, self.homeUser.name))
+                    .then(user => switchUser(plexContext, user, self.homeUser.pin))
+                    .then(extractAuthToken)
+                    .then(token => getAccessToken({token : token, plexApi : plexApi}))
+                    .then(accessToken => {
+                        self.emit('accessToken', accessToken);
+                        self.emit('token', accessToken);
```

Comment:
A plexHome account's access is limited. For example if you were to try to perform a `/library/sections/8/refresh` action it would fail with a `403`. Doing the same action with the regular auth token will work. I keep both of them so if accessToken fails we can fallback to the authToken.